### PR TITLE
Introduce Pipeline Config SPA Stage Settings Modal

### PR DESCRIPTION
**Pull request by GaneshSPatil**
Description:
	- Introduce stage settings modal
	- Define 3 tabs for: 'Stage Settings', 'Environment Variables' & 'Permissions'
	- Implement 'Stage Settings' Widget

Screen Shot 2019-08-23 at 2 43 13 PM

### CODE REVIEW

File: server/webapp/WEB-INF/rails/webpack/views/pages/pipeline_configs/spec/stages/settings/tabs/settings_spec.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  });
+
+  it("should allow toggling fetch materials checkbox", () => {
+    expect((helper.findByDataTestId("form-field-input-fetch-materials")[0] as HTMLInputElement).checked).toEqual(false);
+
+    helper.findByDataTestId("form-field-input-fetch-materials").val("on");
+    simulateEvent.simulate(helper.findByDataTestId("form-field-input-fetch-materials")[0], "click");
+
+    expect((helper.findByDataTestId("form-field-input-fetch-materials")[0] as HTMLInputElement).checked).toEqual(true);
+  });
```
Comment (viraj2712) [inline server/webapp/WEB-INF/rails/webpack/views/pages/pipeline_configs/spec/stages/settings/tabs/settings_spec.tsx]: We need to remove this, as it is duplicate :)

Comment (GaneshSPatil) [reply server/webapp/WEB-INF/rails/webpack/views/pages/pipeline_configs/spec/stages/settings/tabs/settings_spec.tsx]: Not sure if I understood the duplicate code.
Can you point the duplicate piece of code? line?

Comment (viraj2712) [reply server/webapp/WEB-INF/rails/webpack/views/pages/pipeline_configs/spec/stages/settings/tabs/settings_spec.tsx]: https://github.com/gocd/gocd/pull/6814/files#diff-bd2007565d8172520b0dfe850e2ba3beR104-R119

Comment (GaneshSPatil) [reply server/webapp/WEB-INF/rails/webpack/views/pages/pipeline_configs/spec/stages/settings/tabs/settings_spec.tsx]: Hmm. Not sure if I understood the duplicate part.
* `should render fetch materials checkbox` verifies:
    - `fetch-materials` checkbox is rendered on the screen.
    - `fetch-materials`  checkbox is unchecked by default.
* `should allow toggling fetch materials checkbox ` verifies:
   - Initially, the `fetch-materials` checkbox is unchecked by default. And on click, the checkbox state is toggled.

Comment (viraj2712) [reply server/webapp/WEB-INF/rails/webpack/views/pages/pipeline_configs/spec/stages/settings/tabs/settings_spec.tsx]: Sorry for the wrong link

https://github.com/gocd/gocd/pull/6814/files#diff-bd2007565d8172520b0dfe850e2ba3beR87-R119

Comment (GaneshSPatil) [reply server/webapp/WEB-INF/rails/webpack/views/pages/pipeline_configs/spec/stages/settings/tabs/settings_spec.tsx]: Fixed as part of https://github.com/gocd/gocd/pull/6822/files#diff-bd2007565d8172520b0dfe850e2ba3be

**Pull request merged**